### ISSUE TITLE
Fresh Docker install : Import fails silently and button doesn't show up

### ISSUE THREAD

**Issue by FinalDoom** [mylar3/mylar3#881]
Importing my full comics folder "completes" successfully, as far as UI and main logging are concerned, but after initial processing and before the "begin import" (something like that) button shows up, an error shows in the logs.

[carepackage.zip](https://github.com/mylar3/mylar3/files/7149440/carepackage.zip)

The care package probably contains additional logs, this is concerning the error 

> 12-Sep-2021 04:17:51 - INFO    :: mylar.scanLibrary.644 : LibraryScan : [IMPORT] Parsing/Reading of files completed!
12-Sep-2021 04:17:51 - INFO    :: mylar.scanLibrary.645 : LibraryScan : [IMPORT] Attempting to import 4766 files into your watchlist.
12-Sep-2021 04:17:51 - INFO    :: mylar.scanLibrary.646 : LibraryScan : [IMPORT-BREAKDOWN] Files with ComicIDs successfully extracted: 402
12-Sep-2021 04:17:51 - INFO    :: mylar.scanLibrary.647 : LibraryScan : [IMPORT-BREAKDOWN] Files that had to be parsed: 4364
12-Sep-2021 04:17:51 - INFO    :: mylar.scanLibrary.648 : LibraryScan : [IMPORT-BREAKDOWN] Files that were unable to be parsed: 0
12-Sep-2021 04:17:51 - INFO    :: mylar.scanLibrary.649 : LibraryScan : [IMPORT-BREAKDOWN] Files that caused errors during the import: 0
12-Sep-2021 04:22:29 - ERROR   :: mylar.excepthook.315 : LibraryScan : Uncaught exception: Traceback (most recent call last):
  File "/app/mylar3/mylar/logger.py", line 337, in new_run
    old_run(*args, **kwargs)
  File "/usr/lib/python3.8/threading.py", line 870, in run
    self._target(*self._args, **self._kwargs)
  File "/app/mylar3/mylar/librarysync.py", line 698, in scanLibrary
    myDB.upsert("importresults", newValue, controlValue)
  File "/app/mylar3/mylar/db.py", line 173, in upsert
    self.action(query, list(valueDict.values()) + list(keyDict.values()))
  File "/app/mylar3/mylar/db.py", line 130, in action
    sqlResult = self.connection.execute(query, args)
UnicodeEncodeError: 'utf-8' codec can't encode character '\udcfa' in position 11: surrogates not allowed
> 
> 12-Sep-2021 04:22:50 - INFO    :: mylar.future_check.1579 : CP Server Thread-3 : No series have been marked as being on auto-watch.

When choosing a subfolder to do the import against (one that doesn't contain \udcfa apparently), the import completes, and the button shows up. In the screen following clicking the button, all the previously attempted comics show up. Choosing some of the previous ones to import throws a new error, which should be at the end of the log in the care package.

> 12-Sep-2021 04:38:53 - ERROR :: mylar.excepthook.315 : Thread-12 : Uncaught exception: Traceback (most recent call last):
  File "/app/mylar3/mylar/logger.py", line 337, in new_run
    old_run(*args, **kwargs)
  File "/usr/lib/python3.8/threading.py", line 870, in run
    self._target(*self._args, **self._kwargs)
  File "/app/mylar3/mylar/webserve.py", line 4880, in preSearchit
    if 'annual' in getiss.lower():
AttributeError: 'NoneType' object has no attribute 'lower'
> 
> Traceback (most recent call last):
  File "/app/mylar3/mylar/logger.py", line 337, in new_run
    old_run(*args, **kwargs)
  File "/usr/lib/python3.8/threading.py", line 870, in run
    self._target(*self._args, **self._kwargs)
  File "/app/mylar3/mylar/webserve.py", line 4880, in preSearchit
    if 'annual' in getiss.lower():
AttributeError: 'NoneType' object has no attribute 'lower'

My assumption is this error is a result of the prior one.. the import from before should have been cleaned up, but wasn't, allowing a None state object to be accessed. I'm more concerned with how to deal with the encoding error (or how to find the file in question.. possibly something Japanese named, but shouldn't be).

Docker running on Centos, latest build from linuxserver

**Comment by evilhero** [comment]
Well what you need to do is start with [debug mode enabled](https://github.com/mylar3/mylar3/wiki/Debug-logging) so that the full verbose debug logs are available. Enabling this mode prior to do a scan should give alot more information in regards to the files it's scanning in and it might show what the file is with that unknown character.

I just had to google it, but that's an error that's thrown when the tagger you used (I'm assuming it's metatagging) didn't know how to handle a character it came across when writing the metadata, and chunked out that 'unknown character' instead of handling it. I've never come across one personally, so this one is new - if you do happen to find out the file, it'd be interesting to see what was throwing it (metadata, the filename itself, etc).

You're right about the extra files from the previous import session - this is a known bug. You need to browse to **Manage / Advanced Options / Import Results Management** and choose the Flush Imports option in the upper right. This will empty the entries from the table so that you won't get previous sessions (and possible duplicates).

The ``annual in getiss``  error is due to one of the filenames not having a issue number within the filename. This is a limitation of the current version of the importer, and will only accept files that contain an issue number present within the filename (notice I said current version). While Mylar can add / download / manage TPB, GNs and the like with no issue numbers present - the importer was not updated to accommodate the process so atm at least, it cannot handle those files. The prior error actually would have nothing to do with this error as this is common at this juncture and is just in regards to not having an issue number.

The solution is to manually add those series that the issues belong to into Mylar (you can use the CV userscript, or just add in the ComicVine ID for the series into the search box at the top of any page in Mylar and it will add that specific series, as opposed to having to search for it). If you have your folders set up so they line up with the way Mylar creates the directories, it should link up to the appropriate directory you have and see the files therein.

In looking at your logs, you seem to have most tagged - however I also saw you have story arcs listed therein, which is not going to get any results when trying to add into Mylar (it doesn't handle the reading order number), so those would have to be temporarily ignored / moved or at least you're made aware that it won't parse those properly (again just the importer, the rest of Mylar can handle the reading order fine).

The importer is by far the most broken aspect within Mylar. I said `current iteration` earlier because the import has been going thru a rewrite for the past while, but due to lack of time, covid, family, troubleshooting, fixing current bugs amongst everything else - it's still not there yet ( ~75-80% done tho). So I haven't been doing any changes to the importer codebase as that just confused the rewrite further with all the various code changes, which means that these workarounds are the only option atm to initially import items into the system.

**Comment by FinalDoom** [comment]
Hey, thank you for the detailed response.

I enabled debug logging, and while the info around the error doesn't seem any different, there are a lot more logs this time. 
[carepackage(1).zip](https://github.com/mylar3/mylar3/files/7152609/carepackage.1.zip) 

I did find the "Flush Imports" button after submitting this and playing around more. Everything's slowly making more sense/becoming more intuitive.

Grep on the log doesn't show much that stands out to me, hopefully there's something in there that jumps out to you. Let me know anything else I can do to help debug.

> grep -o --color=always -P -h "[\x80-\xFF]" *.log*|sort|uniq
©
á
é
ï
ñ

I'm vaguely remembering I tagged these all with ComicRack a while back.. they've been in backup storage for a few years at this point. 

Is metadata stored in any special way that wouldn't be visible on extract, or is it just the xml file I'm seeing? I can probably write up something to extract all those and find unicode characters.
